odoo/fields.py

```diff
@@ -1921,7 +1921,7 @@ def write(self, records, value):
                 new_store_translations = new_translations
             new_store_translations[lang] = cache_value
 
-            if not records.env['res.lang']._lang_get_id('en_US'):
+            if not records.env['res.lang']._get_data(code='en_US'):
```

Comment:
Wait a minute, I wonder what this boolean test actually does.  If I am not mistaken,
* `records.env['res.lang']._get_data(code='en_US')` returns a (possibly empty) `LangData`, say `lang_data`;
* `not` should evaluate `bool(lang_data)`, which should access `lang_data.__bool__`, which should be caught by method `__getattribute__`, unless Python bypasses this entirely because `lang_data` is an instance of `dict`.

I suggest to add some tests on `LangData` to make sure things like `bool(lang_data)` works as expected.